### PR TITLE
Carousel: clean up touch state after gestures

### LINKED ISSUE #2559: `amp-carousel` overlaps images after rotating device
Steps to reproduce:

1. Go to https://amp.ft.com/content/e96b93ea-2933-11e2-9591-00144feabdc0 on a mobile device, portrait.
2. Swipe a couple of images on the slideshow.
3. Rotate to landscape.
4. Broken layout:
img_0035

NB. this only seems to affect slideshows that are advanced with a swipe gesture. Using the arrow buttons doesn't present a problem with rotation.

Affects at least iOS and Android phones.

Video here: https://dl.dropboxusercontent.com/u/568883/slideshow-overlap.mov

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #2559.

### CODE REVIEW

File: extensions/amp-carousel/0.1/slides.js [hunk 1]
```diff
@@ -182,6 +183,28 @@ export class AmpSlides extends BaseCarousel {
   }
 
   /**
+   * @param {number} index
+   * @private
+   */
+  resetSlide_(index) {
+    const slide = this.slides_[index];
+    if (index == this.currentIndex_) {
+      st.setStyles(slide, {
```
Comment (erwinmombay) [inline extensions/amp-carousel/0.1/slides.js]: @dvoytenko isnt this the same as what `commitSwitch` does?

Comment (erwinmombay) [inline extensions/amp-carousel/0.1/slides.js]: or is this to prepare the `next slide` after the commitSwitch?

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

File: extensions/amp-carousel/0.1/slides.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/slides.js]: No, it's basically the opposite: more like cancel switch.

**Pull request merged**